inst/tests/tests.Rraw

```diff
@@ -9011,59 +9011,60 @@ set.seed(123)
 dt = data.table(x1 = rep(letters[1:2], 6), x2 = rep(letters[3:5], 4), x3 = rep(letters[5:8], 3), y = rnorm(12))
 dt = dt[sample(.N)]
 df = as.data.frame(dt)
+setDT_noattr = function(x) setDT(x, set.allow.inplace.attrib=FALSE)
```

Comment:
IIUC this `setDT_noattr` has been added because `test()` considers `x` and `y` different when one has the new attribute but the other doesn't.  If so, please modify `test()` instead to ignore the new attribute; i.e. add `setattr(xc,"allow.assign.inplace",NULL); setattr(yc,"allow.assign.inplace",NULL)` near line `test.data.table.R:429`. Then revert these test changes, leaving a much smaller set of true test changes.

Reply:
@mattdowle  Sorry, back from a long hiatus. Per your request I reverted the setDT_noattr workaround and modified `test.data.table` logic. It came down to more than a simple additions of `setattr( .., "allow.assign.inplace")` because test() didn't apply this logic to _lists_ of data tables. 
The fix isn't the most general it can be (theoretically it needs to apply for arbitrary nesting of data.tables in lists) but it is good enough for the existing tests.